src/lib.rs

```diff
@@ -160,6 +161,19 @@ impl GlobalStats {
     }
 }
 
+const INTERNAL_KEY_PREFIX: &[u8] = b"__";
+
+pub(crate) fn make_internal_key(k: &[u8]) -> Vec<u8> {
```

Comment:
What if it's used outside the crate?

Reply:
It will behave like a normal key, until:
(1) restart, the key will disappear from memory
(2) rewritten, the key will both memory and disk

Reply:
My point is we should forbid users use such keys.

Reply:
The overhead should be opt-in, if user is unsure he can use `is_internal_key` as check.

Reply:
It's about correctness not about performance. Checking if user writes an internal key is cheap as only prefix (2 bytes) needs to be checked.